src/config_file.c

```diff
@@ -291,7 +291,7 @@ static int config_is_modified(int *modified, struct config_file *file)
 {
 	git_config_file *include;
 	git_buf buf = GIT_BUF_INIT;
-	git_oid hash;
+	git_oid hash = {};
```

Comment:
This fix surprises me. `hash` is only ever being used when `git_hash_buf` returned success, so we should be able to assume that `hash` is always initialized when it is being used. The only possibility I can imagine is that one of our hash implementations is not initializing the hash in certain circumstances, for example when `len == 0`. Which hash implementation were you using?

Reply:
Hmm, not sure. I'd guess whatever's the default.

This was found by running https://github.com/libgit2/libgit2/pull/4433/files#diff-76bfefafc83020b0409259f6bffcc1e9 (which is based off of some of the examples) with the instructions in the accompanying `README.md` and using `memory` as sanitizer. You'll also need a newer version of OpenSSL (1.1k) since they used to have uninitialized memory accesses all over the place in older versions. You might also need to not use the supplied corpus, because for some yet unidentified reason, a call to `opendir` within libFuzzer is triggering an uninitialized memory assertion.